src/ol/source/WMTS.js

```diff
@@ -360,7 +360,7 @@ export default WMTS;
  * @api
  */
 export function optionsFromCapabilities(wmtsCap, config) {
-  const layers = wmtsCap['Contents']['Layer'];
+  const layers = wmtsCap['Contents']['Layer'] || [];
   const l = layers.find(function (elt) {
```

Comment:
Well, if simple, then without the overhead of creating an empty array:
```suggestion
  const layers = wmtsCap['Contents']['Layer'];
  const l = layers?.find(function (elt) {
```